infra/src/test/kotlin/fi/fta/geoviite/infra/tracklayout/LocationTrackDaoIT.kt

```diff
@@ -37,6 +38,10 @@ class LocationTrackDaoIT @Autowired constructor(
     private val locationTrackDao: LocationTrackDao,
     private val layoutDesignDao: LayoutDesignDao,
 ) : DBTestBase() {
+    @BeforeEach
+    fun cleanup() {
+        testDBService.clearAllTables()
+    }
```

Comment:
Nämä testit räjähtivät koska ne yrittivät lisätä uusia designeja samoilla nimillä. Tämä korjaa asian.

Reply:
Tuohonkin pitäisi riittää putsata layout-tablet

Reply:
Riittää joo. Preferoin itse maailmaa jossa putsataan aina ihan koko kanta kaikkien testien välissä (varmistaa varmasti puhtaan tilan, ja yksittäisen IT-testin jäljiltä kannan myös pitäisi olla aina nopeasti putsattavissa vaikka sen kävisikin kokonaan läpi.) Sillä muutin tuon toisenkin, ei sekään aiemmalti hajonnut. Mutta voin muuttaa tämänkin takaisin pelkät layout-taulut putsaavaan muotoon.

Reply:
Joo totta kyllä tuokin. Tavallaan voisi olla ihan fiksukin putsata aina kaikki. En ole varma vaikuttaako se oikeasti paljoakaan, mutta jokainen taulun putsaus on kuitenkin round-trip kantaan ja se voinee kertautua kun testejä on tuhansia.

Reply:
Tuota voisi muuten kokeilla vähän optimoida koska kyllähän tuon putsauksen voisi varmaan toteuttaa niin että ne putsataan kaikki yhdellä round-tripillä. Sitten clearAll olisi varmaan ihan koodin yksinkertaisuudenkin nimissä aina paras :thinking:

Reply:
Totta. Voisin kirjata tästä tiketin

Reply:
Tämän voisi myös toteuttaa niin, että testit luovat uniikit nimet jokaisella suorituskerralla, jolloin kantaa ei tarvitse tyhjentää (ja samalla testi on hieman laajempi, nythän nämä uudet testit menisivät läpi vaikka bisneslogiikkaan olisi kovakoodannut vain pari tiettyä nimeä)

Meillähän on siis oikeastaan useampi testattava tapaus, joista yleensä testeissä on huomioitu vain yksi:
* Testataan kantaan jossa on muutakin (ei välttämättä toimi kantaan joka on tyhjä, vaikka pitäisi)
* Testataan kantaan joka on tyhjä (ei välttämättä toimi kantaan jossa on muutakin, vaikka pitäisi)